### PR TITLE
Export some data from InputAction to the HTTP API

### PR DESCRIPTION
**Pull request by rsandell**
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by rsandell** [comment]
I haven't found why the tests are failing yet, at least on the surface they seem unrelated but very persistent.

**Review by raul-arabaolaza — APPROVED** [review]

**Comment by rsandell** [comment]
Oh, different tests are failing on Ci than locally :scream:

**Comment by rsandell** [comment]
When I fixed one test the other got fixed as well, I still am not sure why, but it works now :shrug:

**Review by raul-arabaolaza — APPROVED** [review]

File: src/test/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepTest.java [hunk 1]
```diff
@@ -611,5 +612,92 @@ public void test_unsafe_ids_generate_formValidation() throws Exception {
         assertThat("> should be rejected", d.doCheckId("this-is-also>-not-ok"), JenkinsMatchers.hasKind(Kind.ERROR));
     }
 
+    @Test
+    public void test_api_contains_waitingForInput() throws Exception {
+        //set up dummy security real
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepTest.java:617]: typo (also probably gratuitous comment)

File: src/test/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepTest.java [hunk 2]
```diff
@@ -611,5 +612,92 @@ public void test_unsafe_ids_generate_formValidation() throws Exception {
...
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        // job setup
+        WorkflowJob foo = j.jenkins.createProject(WorkflowJob.class, "foo");
+        foo.setDefinition(new CpsFlowDefinition(StringUtils.join(Arrays.asList(
+                "def x = input message:'Continue?';",
+                "echo(\"after: ${x}\");"),"\n"),true));
 
+        // get the build going, and wait until workflow pauses
+        QueueTaskFuture<WorkflowRun> q = foo.scheduleBuild2(0);
+        WorkflowRun b = q.getStartCondition().get();
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepTest.java:627]: These can be collapsed using (IIRC) `waitForStart`.

File: src/test/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepTest.java [hunk 3]
```diff
@@ -611,5 +612,92 @@ public void test_unsafe_ids_generate_formValidation() throws Exception {
...
+        JSONObject o = (JSONObject)obj.get();
+        assertTrue(o.has("waitingForInput"));
+        assertTrue(o.getBoolean("waitingForInput"));
+
+        InputAction inputAction = b.getAction(InputAction.class);
+        InputStepExecution is = inputAction.getExecutions().get(0);
+        HtmlPage p = webClient.getPage(b, inputAction.getUrlName());
+        j.submit(p.getFormByName(is.getId()), "proceed");
+
+        json = webClient.getJSON(b.getUrl() + "api/json?depth=1");
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepTest.java:646]: Do not use `depth`. Use `tree` instead.

File: src/test/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepTest.java [hunk 4]
```diff
@@ -611,5 +612,92 @@ public void test_unsafe_ids_generate_formValidation() throws Exception {
...
+                "def chosen = input message: 'Can we settle on this thing?', ok: 'Yep', parameters: [choice(choices: ['Apple', 'Blueberry', 'Banana'], description: 'The fruit in question.', name: 'fruit')], submitter: 'bobby', submitterParameter: 'dd'",
+                "echo(\"after: ${chosen}\");"),"\n"),true));
+
+        // get the build going, and wait until workflow pauses
+        QueueTaskFuture<WorkflowRun> q = foo.scheduleBuild2(0);
+        WorkflowRun b = q.getStartCondition().get();
+        j.waitForMessage("Input requested", b);
+
+        final JenkinsRule.WebClient webClient = j.createWebClient();
+        final JenkinsRule.JSONWebResponse json = webClient.getJSON(b.getUrl() + "api/json?depth=2");
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepTest.java:673]: ditto

**Review by jglick — COMMENTED** [review]
Good to see this!

**Pull request merged**